cpp/unreal_plugins/UrdfBot/Source/UrdfParser.h

```diff
@@ -0,0 +1,175 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#pragma once
+
+#include <map>
+#include <vector>
+
+#include <Eigen/Dense>
+
+#include <CoreMinimal.h>
+
+class FXmlNode;
+
+struct UrdfJointDesc;
+
+enum class UrdfGeometryType
+{
+    Invalid  = 0,
+    Box      = 1,
+    Sphere   = 2,
+    Cylinder = 3,
+    Mesh     = 4,
+};
+
+enum class UrdfJointType
+{
+    Invalid    = 0,
+    Revolute   = 1,
+    Continuous = 2,
+    Prismatic  = 3,
+    Fixed      = 4,
+    Floating   = 5,
+    Planar     = 6
+};
+
+enum class CalibrationType
+{
+    Invalid = 0,
+    Rising  = 1,
+    Falling = 2
+};
+
+struct UrdfGeometryDesc
+{
+    UrdfGeometryType type_;
+    FVector size_ = FVector::ZeroVector; // required by Box, in [m]
+    float radius_ = 0.0f; // required by Sphere, Cylinder, in [m]
+    float length_ = 0.0f; // required by Cylinder, in [m]
+    std::string filename_; // required by Mesh
+    float scale_ ; // optional for Mesh
+};
+
+struct UrdfMaterialDesc
```

Comment:
```
struct UrdfMaterialDesc
{
    std::string name_;

    FVector4 color_ = FVector4::ZeroVector;
    std::string texture_;

    // derived data
    bool is_reference_               = false;
    UrdfMaterialDesc* material_desc_ = nullptr;
};
```

Reply:
FVector4 has default value (0,0,0,1) and does not have a ZeroVector.

Reply:
Ah ok that makes sense. rgba=0,0,0,1 is a better default.